### ISSUE TITLE
Add re-export of nom's `IResult`

### ISSUE THREAD

**Issue by jonhoo** [djc/imap-proto#1]
Given that `IResult` is returned by `parse_response`, it would be useful for `imap-proto` to re-export `nom::IResult` (as well as perhaps the various types included in its variants) so that users of this crate do not need to explicitly pull in `nom`.

**Comment by djc** [comment]
Yeah -- I experimented with exporting a type alias called `ParserResult` to abstract over the `nom` dependency a little bit, but that did not seem to work so well because the enum variants don't seem to be accessible through the alias. They really should be to make it ergonomic to consume the result, so I'll need to do a bit more experimentation.

**Comment by djc** [comment]
I opened a Rust internals thread about this topic: https://internals.rust-lang.org/t/pre-rfc-enum-variants-through-type-aliases/6433.
